### PR TITLE
feat(dialog): add experimental dialog

### PR DESCRIPTION
**Pull request by alatielle**
## What

Adds an experimental Dialog Component using the new Modal & Backdrop components. I decided to not re-export them as public yet

### Media

​
image

## Why

​https://jira.intapps.it/browse/DISP-331

### CODE REVIEW

File: src/components/experimental/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const ButtonsWrapper = styled.div`
+    padding-top: 1.5rem;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    gap: 1rem;
+`;
+
+const StyledModal = styled(Modal)`
+    width: 30rem;
```
Comment (lloydaf) [inline src/components/experimental/Dialog/Dialog.tsx:23]: shouldn't this be controllable?

Comment (alatielle) [reply src/components/experimental/Dialog/Dialog.tsx:23]: In the dialog -- no, in a modal -- absolutely!

File: src/components/experimental/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+interface DialogProps extends Omit<BackdropProps, 'isDismissable' | 'isKeyboardDismissDisabled'> {
+    role?: 'dialog' | 'alertdialog';
+    headline: ReactNode;
+    subtitle: ReactNode;
+    dismissButton: ReactNode;
+    actionButton: ReactNode;
+}
+
+const Dialog = ({
+    role = 'dialog',
```
Comment (lloydaf) [inline src/components/experimental/Dialog/Dialog.tsx:44]: Should we allow other roles?

File: src/components/experimental/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    dismissButton,
+    actionButton,
+    ...props
+}: DialogProps): ReactElement => (
+    <Backdrop {...props} isDismissable={false} isKeyboardDismissDisabled>
+        <StyledModal role={role}>
+            <Card>
+                <HeadlineText slot="title">{headline}</HeadlineText>
+
+                <SubtitleText as="p" variant="body1">
```
Comment (lloydaf) [inline src/components/experimental/Dialog/Dialog.tsx:56]: Maybe not within the scope of the PR, but can we name the variants with a more semantic name? `body1`, `body2` etc. don't help me understand what the variant does

File: src/components/experimental/Dialog/Dialog.tsx (hunk 2 above)
Comment (alatielle) [reply src/components/experimental/Dialog/Dialog.tsx:44]: We do, It's either dialog or alertdialog

File: src/components/experimental/Dialog/Dialog.tsx (hunk 3 above)
Comment (alatielle) [reply src/components/experimental/Dialog/Dialog.tsx:56]: It's named like the designers name it and what they use in the designs

**Review by dornelasnelson — APPROVED** [review]

**Pull request merged**